src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,216 @@
+                    break;
+                case "unmark":
+                    Task unmarked = this.taskList.changeTaskStatus(Integer.parseInt(infoArray[1]), false);
+                    this.ui.printChangeTaskStatus(unmarked, false);
+                    break;
+                case "delete":
+                    Task deleted = this.taskList.deleteTask(Integer.parseInt(infoArray[1]));
+                    this.ui.printDeleteTask(deleted, this.taskList);
+                    break;
+                case "find":
+                    ArrayList<Task> targetTasks = this.taskList.findMatchingTasks(infoArray[1]);
+                    this.ui.printMatchingTasks(targetTasks);
+                    break;
+                case "todo":
+                    Task todo = new Todo(infoArray[1]);
+                    taskList.addTask(todo);
+                    this.ui.printAddTask(todo, this.taskList);
+                    break;
+                case "deadline":
+                    Task deadline = new Deadline(
+                            infoArray[1],
+                            LocalDateTime.parse(infoArray[2], dateTimeFormatter)
+                    );
+                    taskList.addTask(deadline);
+                    this.ui.printAddTask(deadline, this.taskList);
+                    break;
+                case "event":
+                    Task event = new Event(
+                            infoArray[1],
+                            LocalDateTime.parse(infoArray[2], dateTimeFormatter),
+                            LocalDateTime.parse(infoArray[3], dateTimeFormatter)
+                    );
+                    taskList.addTask(event);
+                    this.ui.printAddTask(event, this.taskList);
+                    break;
+                default:
+                    throw new DukeException("I'm sorry but I don't know what that means!");
+                }
+            } catch (DukeException e) {
+                this.ui.printDukeException(e);
+            } catch (DateTimeParseException e) {
+                this.ui.printDateTimeParseException();
+            }
+
+            currentLine = this.ui.nextLine();
+        }
+
+        try {
+            this.storage.writeTasksToStorage(this.taskList);
+        } catch (IOException e) {
+            this.ui.printIoException(e);
+        }
+
+        this.ui.printDukeClosing();
+        this.ui.closeScanner();
+        System.exit(0);
+    }
+
+    /**
+     * Method for the GUI Duke Program.
+     *
+     * @param input The input text from which to get a response.
+     * @return Returns the properly formatted response to display on the GUI.
+     */
+    public String getResponse(String input) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        String response = "";
+        try {
+            String information = Parser.parseCommand(input);
+            String[] infoArray = information.split("\\|");
+
+            switch (infoArray[0]) {
```

Comment:
Will it be clearer if you declare another variable for `infoArray[0]` to clarify the meaning of it? In this case, I guess `infoArray[0]` is a command, right?

Reply:
That's a good suggestion 👍 . Thank you, I can update that!